rest/src/main/java/tennisclub/controllers/UserController.java

```diff
@@ -1,13 +1,59 @@
 package tennisclub.controllers;
 
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import tennisclub.dto.user.UserAuthDTO;
+import tennisclub.dto.user.UserFullDTO;
+import tennisclub.dto.user.UserUpdateDTO;
+import tennisclub.entity.User;
+import tennisclub.facade.UserFacade;
 
+import java.util.List;
+
+/**
+ * Controller for manipulation with users
+ * @author Ondrej Holub
+ */
+@CrossOrigin
 @RestController
+@RequestMapping("/users")
 public class UserController {
 
-    @GetMapping("/test")
-    public final String testConnection() {
-        return "This works";
+    private final UserFacade userFacade;
+
+    @Autowired
+    public UserController(UserFacade userFacade) {
+        this.userFacade = userFacade;
+    }
+
+    @PostMapping
+    public final UserFullDTO register(@RequestBody UserAuthDTO userAuthDTO) {
+        return userFacade.register(userAuthDTO);
+    }
+
+    @GetMapping("/auth")
+    public final boolean authenticate(@RequestBody UserAuthDTO userAuthDTO) {
+        return userFacade.authenticate(userAuthDTO);
+    }
```

Comment:
Also, a `GET` request should never have a body. [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231) does not explicitly forbid it, but in practice:
- it might not be implemented by servers and clients (e.g. Angluar's `HttpClient` does not accept a body in the shorthand `.get()` method);
- to quote [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET):
  > Note: Sending body/payload in a GET request may cause some existing implementations to reject the request — while not prohibited by the specification, the semantics are undefined. It is better to just avoid sending payloads in GET requests.

Therefore this should be a `POST` endpoint.

**Note:** The `POST` method in the context of REST should usually be used for *adding a resource to a collection*. However, in practice, it is **also** used for general RPC-like endpoints which do not match any of the usual semantics of `GET`, `POST`, `PUT`, ... methods. An authentication/authorization endpoint typically neither modifies, nor returns the state of an entity, but rather performs an auxiliary action on the system as a whole, so it is exactly the aforementioned case.